src/Blockcore.Indexer.Cirrus/Storage/Mongo/SmartContracts/DaoContractAggregator.cs

```diff
@@ -0,0 +1,138 @@
+      this.logger = logger;
+
+      cirrusClient = clientFactory.Create(connection) as CirrusClient;
+   }
+
+   public async Task<DaoContractComputedTable> ComputeDaoContractForAddressAsync(string address)
+   {
+      var contract = await LookupDaoContractForAddressAsync(address);
+
+      if (contract is null)
+         return null;
+
+      var contractTransactions = await db.CirrusContractTable
+         .AsQueryable()
+         .Where(_ => _.ToAddress == address && _.Success && _.BlockIndex > contract.LastProcessedBlockHeight)
+         .ToListAsync();
+
+      if (contractTransactions.Any())
+      {
+         await AddNewTransactionsDataToDocumentAsync(address, contractTransactions, contract);
+      }
+
+      return contract;
+   }
+
+   async Task<DaoContractComputedTable> LookupDaoContractForAddressAsync(string address)
+   {
+      DaoContractComputedTable contract = await db.DaoContractComputedTable
+         .AsQueryable()
+         .SingleOrDefaultAsync(_ => _.ContractAddress == address);
+
+      if (contract is not null)
+         return contract;
+
+      var contractCode = await db.CirrusContractCodeTable
+         .AsQueryable()
+         .SingleOrDefaultAsync(_ => _.ContractAddress == address);
+
+      if (contractCode is null || contractCode.CodeType != DaoContract)
+      {
+         logger.LogInformation(
+            $"Request to compute DAO contract for address {address} which was not found in the contract code table");
+         return null;
+      }
+
+      contract = await CreateNewDaoContract(address);
+
+      if (contract is null)
+         throw new ArgumentNullException($"Contract not found in the contract table for address {address}");
+
+      return contract;
+   }
+
+   private async Task<DaoContractComputedTable> CreateNewDaoContract(string address)
+   {
+      var contractCreationTransaction = await db.CirrusContractTable
+         .AsQueryable()
+         .Where(_ => _.NewContractAddress == address)
+         .SingleOrDefaultAsync();
+
+      if (contractCreationTransaction is null)
+         throw new ArgumentNullException(nameof(contractCreationTransaction));
+
+      var contract = new DaoContractComputedTable
+      {
+         ContractAddress = contractCreationTransaction.NewContractAddress,
+         ContractCreateTransactionId = contractCreationTransaction.TransactionId,
+         LastProcessedBlockHeight = contractCreationTransaction.BlockIndex
+      };
+
+      await SaveTheContractAsync(address, contract);
+
+      return contract;
+   }
+
+   private async Task AddNewTransactionsDataToDocumentAsync(string address, List<CirrusContractTable> contractTransactions,
+      DaoContractComputedTable contract)
+   {
+      foreach (var contractTransaction in contractTransactions)
+      {
+         var reader = logReaderFactory.GetLogReader(contractTransaction.MethodName);
+
+         if (reader is null)
+         {
+            Console.WriteLine(contractTransaction.MethodName);
```

Comment:
Don't use `Console.WriteLine` we have an ILog no? 
maybe instead we should log a warning?